### PR TITLE
Add support for the CHECK constraint

### PR DESCRIPTION
**Pull request by tyamagu2**
closes https://github.com/makenowjust/memefish/issues/38

### CODE REVIEW

File: pkg/ast/ast.go [hunk 1]
```diff
@@ -1263,7 +1263,7 @@ type CreateTable struct {
 	Name              *Ident
 	Columns           []*ColumnDef
 	PrimaryKeys       []*IndexKey
-	ForeignKeys       []*ForeignKey
+	TableConstraints  []*TableConstraint       // optional
```
Comment (makenowjust) [inline pkg/ast/ast.go:1266]: I'm not sure this `// optional` comment is necessary. Contrary, there is `// len(Whens) > 0` comment, so we assumed all slice fields without comment contain an empty slice.

File: pkg/ast/ast.go [hunk 2]
```diff
@@ -1404,16 +1434,14 @@ type AddColumn struct {
 	Column *ColumnDef
 }
 
-// AddForeignKey is ADD FOREIGN KEY clause in ALTER TABLE.
-//
-//     ADD {{.ForeignKey | sql}}
-type AddForeignKey struct {
+// AddTableConstraint is ADD table_constraint clause in ALTER TABLE.
```
Comment (makenowjust) [inline pkg/ast/ast.go:1437]:
```suggestion
// AddTableConstraint is ADD table_constraint clause in ALTER TABLE.
//
//     ADD {{.TableConstraint}}
```

**Review by makenowjust — CHANGES_REQUESTED** [review]
Implementation seems good. I was curious about some comments. Please see above.

File: pkg/ast/ast.go (hunk 1 above)
Comment (tyamagu2) [inline pkg/ast/ast.go:1266]:
```suggestion
	TableConstraints  []*TableConstraint
```

Comment (tyamagu2) [reply pkg/ast/ast.go]: OK, makes sense.

**Review by makenowjust — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**